test/models/katello/alternate_content_source_test.rb

```diff
@@ -115,6 +115,24 @@ def test_with_type
       @simplified_acs.save!
       assert_equal [@yum_acs, @rhui_acs, @simplified_acs].sort, AlternateContentSource.with_type('yum').sort
     end
+
+    def test_update_acs_type
+      assert_raises(ActiveRecord::RecordInvalid) do
+        @simplified_acs.update!(alternate_content_source_type: "custom")
+      end
+      assert_raises(ActiveRecord::RecordInvalid) do
+        @rhui_acs.update!(alternate_content_source_type: "simplified")
+      end
+    end
+
+    def test_update_content_type
+      assert_raises(ActiveRecord::RecordInvalid) do
```

Comment:
I don't think it's strictly necessary, but you can match the error message too:

```ruby
        exception = assert_raises(RuntimeError) do
          plan_action(action, organization: organization, path: path, metadata: metadata)
        end
        assert_match(/'#{content_view_version.name}' already exists/, exception.message)
```

Reply:
I like things like this! This way if the record is invalid for an unrelated reason, the test will still do what it's supposed to.